### PR TITLE
Fix gcc compile errors with caused by sign checks

### PR DESCRIPTION
**Pull request by antonrohr**
This is affecting OrbitSsh::Socket and OrbitSsh::Channel

### CODE REVIEW

**Comment by florian-kuebler** [comment]
Just one question: Is there in general any reason to use `int` here  instead of `uint64_t`, or `uint32_t`, `size_t`

File: OrbitSsh/Channel.cpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ ResultType Channel::Read(std::string* result, int buffer_size) {
   return ResultType::kAgain;
 }
 
-ResultType Channel::Write(const std::string& text, int* written_length) {
+ResultType Channel::Write(const std::string& text, uint* written_length) {
```
Comment (dimitry-) [inline OrbitSsh/Channel.cpp:119]: yeah, you probably want size_t here... and below

Comment (antonrohr) [reply OrbitSsh/Channel.cpp]: true, thank you

**Comment by antonrohr** [comment]
@florian-kuebler I don' think there is a good reason. I changed it to `size_t`

**Pull request merged**